File: backend/app/routers/ai.py
from fastapi import APIRouter
from ..schemas import ChatIn, CropSuggestIn
from ..services.ai_chat import chat as chat_impl
from ..services.ai_crops import suggest as crop_suggest

router = APIRouter(tags=["ai"])

@router.post("/ai/chat")
def chat(body: ChatIn):
    return {"reply": chat_impl(body.message)}

@router.post("/ai/crops")
def crops(body: CropSuggestIn):
    items = crop_suggest(body.region, body.season, body.soil, body.marketDemand, body.cropType)
    return items

File: frontend/src/services/ai.js
// AI services wired to backend FastAPI
export async function getCropSuggestions({ region, season, soil, marketDemand, cropType }) {
  try {
    const res = await fetch('/api/ai/crops', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ region, season, soil, marketDemand, cropType })
    })
    if (!res.ok) throw new Error(await res.text())
    return await res.json()
  } catch (e) {
    console.warn('getCropSuggestions failed, falling back to local rules', e)
    const base = [
      { crop: 'Wheat', reason: 'Favorable winter temperatures and good market demand.' },
      { crop: 'Paddy', reason: 'Sufficient water availability and traditional region fit.' },
      { crop: 'Millets', reason: 'Climate-resilient and high nutritional demand.' },
    ]
    return base.filter((_, i) => (marketDemand ? i !== 1 : true))
  }
}

export async function chatWithBot(message) {
  try {
    const res = await fetch('/api/ai/chat', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ message })
    })
    if (!res.ok) throw new Error(await res.text())
    const data = await res.json()
    return data.reply || ''
  } catch (e) {
    console.warn('chatWithBot failed, using fallback', e)
    return `AI: For your query "${message}", consider soil testing and local mandi prices.`
  }
}
